### PR TITLE
Enable redis password by default

### PR DESCRIPTION
**Pull request by simon-mo**
- Randomly generate 16 alphanumerics. 
- dashboard, ray.init('auto') continue to work
- only generate password when user run `ray.init()` in python
- not the autoscaler.

### CODE REVIEW

**Comment by ericl** [comment]
Won't this be unnecessary if we bind to 127.0.0.1 in local mode anyways? Why not hard code a password for now to protect both cases?

File: python/ray/scripts/scripts.py [hunk 1]
```diff
@@ -252,10 +252,14 @@ def start(node_ip_address, redis_address, address, redis_port,
         node_ip_address = services.address_to_ip(node_ip_address)
 
     if redis_address is not None or address is not None:
-        (redis_address, redis_address_ip,
-         redis_address_port) = services.validate_redis_address(
+        (redis_address, redis_address_ip, redis_address_port,
+         redis_password_found) = services.validate_redis_address(
```
Comment (richardliaw) [inline python/ray/scripts/scripts.py]: In what situation will I have a redis_password that I find here?

Like, I can only think of two cases:
1. redis is not started on the node yet, in which case there is no password.
2. redis is started on the node, in which case I die.

**Comment by richardliaw** [comment]
Skimmed the code but will defer the conversation whether this is necessary to @ericl and @simon-mo.

**Comment by simon-mo** [comment]
Closed for #6481

**Pull request closed without merging**